pkg/database/database.go

```diff
@@ -7,22 +7,26 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
-	"log"
-	"time"
 
-	"github.com/XSAM/otelsql"
-	"github.com/avast/retry-go/v4"
 	"github.com/kubearchive/kubearchive/pkg/models"
-	_ "github.com/lib/pq"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-const (
-	resourceTableName        = "resource"
-	resourcesQuery           = "SELECT data FROM %s WHERE kind=$1 AND api_version=$2"
-	namespacedResourcesQuery = "SELECT data FROM %s WHERE kind=$1 AND api_version=$2 AND namespace=$3"
-	writeResource            = `INSERT INTO %s (uuid, api_version, kind, name, namespace, resource_version, cluster_deleted_ts, data) Values ($1, $2, $3, $4, $5, $6, $7, $8) ON CONFLICT(uuid) DO UPDATE SET name=$4, namespace=$5, resource_version=$6, cluster_deleted_ts=$7, data=$8`
-)
+type newDatabaseFunc func(map[string]string) DBInterface
+
+var RegisteredDatabases = make(map[string]newDatabaseFunc)
+
+type DBInfoInterface interface {
+	GetDriverName() string
+	GetConnectionString() string
+	GetResourcesSQL() string
+	GetNamespacedResourcesSQL() string
+	GetWriteResourceSQL() string
+}
+
+type DatabaseInfo struct {
+	env map[string]string
+}
```

Comment:
I believe there is no need for this struct anymore
```suggestion
```

Reply:
Fixed.